app/services/post_status_service.rb

```diff
@@ -39,15 +39,16 @@ def validate_media!(media_ids)
 
     raise Mastodon::ValidationError, I18n.t('media_attachments.validations.too_many') if media_ids.size > 4
 
-    media = MediaAttachment.where(status_id: nil).where(id: media_ids.take(4).map(&:to_i))
+    target_media_ids = media_ids.take(4).map(&:to_i)
+    media = target_media_ids.blank? ? [] : MediaAttachment.where(status_id: nil).where(id: target_media_ids)
```

Comment:
Would it be possible to just set media to nil here, instead of an empty array, and avoid the change lower down?